store/tikv/region_request.go

```diff
@@ -712,6 +735,13 @@ func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed
 		logutil.BgLogger().Warn("tikv reports `RaftEntryTooLarge`", zap.Stringer("ctx", ctx))
 		return false, errors.New(regionErr.String())
 	}
+	// A stale read request may be sent to a peer which has not been initialized yet, we should retry in this case.
+	if regionErr.GetRegionNotInitialized() != nil {
+		logutil.BgLogger().Warn("tikv reports `RegionNotInitialized` retry later",
+			zap.Uint64("region-id", regionErr.GetRegionNotInitialized().GetRegionId()),
+			zap.Stringer("ctx", ctx))
+		return true, nil
+	}
```

Comment:
I think we can store the `isStaleRead` in `rpcContext`, then we can know whether this is a stale read request here.

Reply:
Also we should consider `DataNotReady` error

Reply:
Done.